src/screens/MainApp/DiagnoseRequest/navigators/MainNavigator.js

```diff
@@ -22,21 +22,15 @@ const Navigator = buildStackNavigator(
 
 const MainNavigator = createAppContainer(Navigator)
 
+const allowedRoutesToShowTabBar = ['MainScreen', 'NoPhotoDisclaimer', 'FinishRequest']
+
 MainNavigator.navigationOptions = ({ navigation }) => {
-  let tabBarVisible
-  const allowedRoutesToShowTabBar = ['MainScreen', 'NoPhotoDisclaimer', 'FinishRequest']
+  if (navigation.state.routes.length <= 1) return {}
 
-  if (navigation.state.routes.length > 1) {
-    navigation.state.routes.map(route => {
-      if (includes(allowedRoutesToShowTabBar, route.routeName)) {
-        tabBarVisible = true
-      } else {
-        tabBarVisible = false
-      }
-    })
-  }
+  const routesInState = navigation.state.routes.map(route => route.routeName)
+  const tabBarVisible = includes(allowedRoutesToShowTabBar, routesInState.pop())
```

Comment:
I made a similar comment about this in another PR. https://github.com/WeDevelop-ARG/cannabis-app/pull/96#discussion_r376469850